client/my-sites/plans-v2/plans-filter-bar/index.tsx

```diff
@@ -34,12 +44,48 @@ interface Props {
 	duration?: Duration;
 	productType?: ProductType;
 	onDurationChange?: ( arg0: Duration ) => void;
-	onProductTypeChange?: ( arg0: Duration ) => void;
+	onProductTypeChange?: ( arg0: ProductType ) => void;
 }
 
 const CALYPSO_MASTERBAR_HEIGHT = 47;
 const CLOUD_MASTERBAR_HEIGHT = 94;
 
+const REALTIME_PRODUCTS = [ PLAN_JETPACK_SECURITY_REALTIME, PRODUCT_JETPACK_BACKUP_REALTIME ];
+
+const DiscountMessage = () => {
+	const translate = useTranslate();
+
+	const croVersion = getJetpackCROActiveVersion();
+	const slugsToCheck = [
+		...JETPACK_PRODUCTS_BY_TERM.map( ( s ) => s.yearly ),
+		...JETPACK_RESET_PLANS_BY_TERM.map( ( s ) => s.yearly ),
+	].filter( ( slug ) => {
+		// Don't factor in real-time products for CRO v1 --
+		// they're not visible in the product grid
+		if ( croVersion === 'v1' && REALTIME_PRODUCTS.includes( slug ) ) {
+			return false;
+		}
+
+		return true;
+	} );
+	const highestAnnualDiscount = useSelector( ( state ) =>
+		getHighestAnnualDiscount( state, slugsToCheck )
+	);
+
+	if ( ! highestAnnualDiscount ) {
+		return null;
+	}
+
+	return (
+		<span className="plans-filter-bar__discount-message">
+			{ translate( 'You save up to %(discount)s by paying yearly', {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'You save %(discount)s by paying yearly' )` *ES Score: 7*